### PR TITLE
[Harness] Move the definition of the tasks outside of the jenkins class.

### PR DESCRIPTION
**Pull request by mandel-macaque**
Create some enumerables that will state the tests to run, that way, if
new tests need to be added, there will not be a need to edit the
jenkins class.

Add tests to ensure that all the tests are added. That away we will make
sure that tests are not removed by accident.

### CODE REVIEW

File: tests/xharness/Jenkins/Jenkins.cs [hunk 1]
```diff
@@ -447,7 +447,7 @@ Task<IEnumerable<AppleTestTask>> CreateRunDeviceTasksAsync ()
 		}
 
 		public bool IsBetaXcode => Harness.XcodeRoot.IndexOf ("beta", StringComparison.OrdinalIgnoreCase) >= 0;
-		
+
```
Comment (rolfbjarne) [inline tests/xharness/Jenkins/Jenkins.cs:450]: Whitespace changes

File: tests/xharness/Jenkins/MacTestTasksFactory.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+namespace Xharness.Jenkins {
+	class MacTestTasksFactory : IEnumerable<RunTestTask> {
+
+		readonly Jenkins jenkins;
+		readonly IProcessManager processManager;
+		readonly ICrashSnapshotReporterFactory crashReportSnapshotFactory;
+		readonly ITestVariationsFactory testVariationsFactory;
+
+		public MacTestTasksFactory (Jenkins jenkins, 
+								    IProcessManager processManager,
```
Comment (rolfbjarne) [inline tests/xharness/Jenkins/MacTestTasksFactory.cs:19]: Indentation looks funny.

File: tests/xharness/Jenkins/Jenkins.cs (hunk 1 above)
Comment (mandel-macaque) [reply tests/xharness/Jenkins/Jenkins.cs:450]: VSMac has its own will lately

File: tests/xharness/Jenkins/MacTestTasksFactory.cs (hunk 2 above)
Comment (mandel-macaque) [reply tests/xharness/Jenkins/MacTestTasksFactory.cs:19]: I think is GitHub getting confused, look ok in my editor, will double check.

File: tests/xharness/Jenkins/Jenkins.cs [hunk 3]
```diff
@@ -459,210 +459,24 @@ Task PopulateTasksAsync ()
...
-			};
-			Tasks.Add (nunitExecutioniOSMSBuild);
-
-			var installSourcesProject = new TestProject (Path.GetFullPath (Path.Combine (Harness.RootDirectory, "..", "tools", "install-source", "InstallSourcesTests", "InstallSourcesTests.csproj")));
-			var buildInstallSources = new MSBuildTask (jenkins: this, testProject: installSourcesProject, processManager: processManager)
-			{
-				SpecifyPlatform = false,
-				SpecifyConfiguration = false,
-				Platform = TestPlatform.iOS,
+			// al factories are enumberators \o/
```
Comment (spouliot) [inline tests/xharness/Jenkins/Jenkins.cs:467]: typo `all`

**Pull request merged**